app/src/main/java/ar/com/wolox/android/example/ui/example/login/LoginFragment.java

```diff
@@ -106,4 +108,25 @@ private void saveLoginDataOnSharedPreferences(String email, String password) {
             .putBoolean(getString(R.string.shared_preference_login_status), true)
             .apply();
     }
+
+    @Override
+    public void onInvalidLoginData() {
+        Toast toast = Toast.makeText(getContext(), getString(
+                R.string.fragment_login_invalid_email_or_password_toast), Toast.LENGTH_LONG);
+        toast.show();
+    }
+
+    @Override
+    public void onServerError() {
+        Toast toast = Toast.makeText(getContext(), getString(
```

Comment:
This is code ok, however I would like to use some dependency injection here. Can you change it?

Reply:
✔ **Done**

Now I understand and love injection :smile_cat: